pkg/util/own_resources.go

```diff
@@ -160,6 +160,9 @@ func getDeploymentFromPod(pod *corev1.Pod, c client.Reader, operatorNs string, l
 }
 
 func getCSVFromDeployment(d *appsv1.Deployment, c client.Reader, operatorNs string, logger logr.Logger) (*csvv1alpha1.ClusterServiceVersion, error) {
+	if d == nil {
+		return nil, nil
+	}
```

Comment:
Added this check because of the panicking unit tests for `pkg/webhooks/validator/validator_test.go`.

Reply:
The way I came to make this change was by running the test through GoLand and stepping through the code. It was panicking at the `d.GetOwnerReferences()` in below: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a3e3030b3f9d0dc7000ff603131adb68f1e1c4f0/pkg/util/own_resources.go#L162-L164 

So I added a `nil` check. The call originates from https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a3e3030b3f9d0dc7000ff603131adb68f1e1c4f0/pkg/webhooks/validator/validator_test.go#L1500

I don't claim to know this 100%, but it doesn't look like fake client is being used here.

Separately, a similar `nil` check exists for another function in the same file: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a3e3030b3f9d0dc7000ff603131adb68f1e1c4f0/pkg/util/own_resources.go#L119-L122

Reply:
> I don't claim to know this 100%, but it doesn't look like fake client is being used here.

please see:
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a3e3030b3f9d0dc7000ff603131adb68f1e1c4f0/pkg/webhooks/validator/validator_test.go#L1439

Reply:
I meant that the line in the validator_test.go file that's panicking doesn't seem to me like a direct call to the client. It's calling some util function which is working on a (nil) pointer to a Deployment

Reply:
Even though a fake client is being passed, the Deployment resource, which is the cause of the panic, doesn't look related to the fake client. But I could be wrong. :)